Completion/Unix/Command/_find

```diff
@@ -126,7 +129,7 @@ _arguments -C $args \
   '*-atime:access time (days):->times' \
   '*-ctime:inode change time (days):->times' \
   '*-depth' \
-  '*-exec:program: _command_names -e:*\;::program arguments: _normal' \
+  '*-exec:program: _command_names -e:*(\;|+)::program arguments: _normal' \
```

Comment:
This part does seem to have the effect of causing `find . -exec ls \; -<TAB>` to offer find(1) options rather than ls(1) options, but I can't get it to _offer_ `\;` or `+` as a completion, and in any case it seems to be relying on undocumented behaviour of `_arguments`.  (That's a preëxisting issue.)

For reference, this line is an `_arguments` optspec for the `-exec` option, with a leading `*` meaning the option can repeated, and with completion for multiple following words (see "This group may be repeated" in the documentation of `_arguments`).  The part `*\;:` is supposed to be another optspec, but neither `\;` nor `(\;|+)` is a valid optspec.

Reply:
So what is the right way to get it to work for + and ;?

Reply:
I'm not sure what you mean is undocumented. Quoting the zsh documentation:
> This describes multiple arguments. Only the last optarg for an option taking multiple arguments may be given in this form. If the pattern is empty (i.e. :*:), all the remaining words on the line are to be completed as described by the action; otherwise, all the words up to and including a word matching the pattern are to be completed using the action.

Also, competing arguments for ls seem to work just fine for me?  I cannot replicate the issue you describe. That is using this file along with the latest release of zsh (5.8.1). I don't have zsh itself built from git. Maybe something else broke recently?

Reply:
I stand corrected: it's indeed documented as you say.

What I couldn't get to work is for `find . -exec ls <TAB>` to offer `\;` as a potential completion.  That's not a regression though so let's not worry about that.

Reply:
I stand corrected: it's indeed documented as you say.

What I couldn't get to work is for `find . -exec ls <TAB>` to offer `\;` as a potential completion.  That's not a regression though so let's not worry about that.

Reply:
> What I couldn't get to work is for find . -exec ls <TAB> to offer \; as a potential completion. That's not a regression though so let's not worry about that.

Ah, now I see what you mean. I guess you could use (I haven't tested this!) `{_alternatives '_values \; +' _normal}` or something to that effect (but with valid syntax)

Reply:
Or perhaps with `_arguments` states and/or `_wanted terminator expl terminators compadd \; +`.

Anyway, as I said, it's not a regression.  If you get this to work, great; if not, it's not a blocker to merging the rest.

Reply:
> ` _wanted terminator expl terminators compadd \; +.`

I have no idea what you just wrote. I clearly have a lot to learn still. ZSH argument completion is... dense. And so is the documentation. And most tutorials online only cover the basics. I really don't understand what _wanted is all about to be honest.

(Side note: I would love to see some more approachable documentation for the completion system as a whole. As it is now it is very dense reference documentation. Just adding examples of where you would want to use each feature (exists in some cases, but not all) would help a lot for example. Sure I can look at one of the existing completion files, but they are generally complex beasts, not made specifically to demonstrate a feature. Nor do they have the code comments that would be required in an illustrative example. Also `_arguments` and the other "big" ones would benefit from being restructured to have more sub-headings to split the documentation up so it is easier to find things in.)

Reply:
> I would love to see some more approachable documentation for the completion system as a whole.

There are the [zsh-completions how-to](https://github.com/zsh-users/zsh-completions/blob/master/zsh-completions-howto.org) and the [style guide](https://github.com/zsh-users/zsh/blob/master/Etc/completion-style-guide). Though even those aren't great imo. I had started writing my own guide at one point, but there's so much to cover and i tend to over-explain things so i worried that it'd just end up like the manual again...

(I haven't really looked at any of the changes or technical discussion here btw)

Reply:
@okdana I'd be happy to take a look at your draft, time permitting.  Feel free to send me a link (publicly or privately).